packages/interbit-ui-tools/src/tests/middleware/sagas.test.js

```diff
@@ -0,0 +1,588 @@
+          payload: { error: errorMessage }
+        }
+      })
+      .not.put.like({
+        action: { type: actionTypes.STATIC_CHAINS_SAGA }
+      })
+      .run()
+  })
+
+  it('If no static chains are configured, no problem', async () => {
+    const contextWithoutStaticChains = mockContext({
+      config: configWithoutStaticChains
+    })
+    const { rootSaga } = middlewareSagas(contextWithoutStaticChains)
+    await expectSaga(rootSaga)
+      .withReducer(reducers)
+      .put.like({
+        action: {
+          type: actionTypes.INITIAL_CONFIG,
+          payload: configWithoutStaticChains
+        }
+      })
+      .put.like({
+        action: { type: actionTypes.STATIC_CHAINS_SAGA }
+      })
+      .not.put.like({
+        action: {
+          type: actionTypes.INTERBIT_ERROR
+        }
+      })
+      .not.put.like({
+        action: {
+          type: actionTypes.CHAIN_ERROR
+        }
+      })
+      .run()
+  })
+
+  it('If static chains are configured, load them', async () => {
+    const contextWithStaticChains = mockContext({
+      config: configWithStaticChains
+    })
+    const { rootSaga } = middlewareSagas(contextWithStaticChains)
+    await expectSaga(rootSaga)
+      .withReducer(reducers)
+      .put.like({
+        action: {
+          type: actionTypes.INITIAL_CONFIG,
+          payload: configWithStaticChains
+        }
+      })
+      .put.like({
+        action: {
+          type: actionTypes.CHAIN_LOADING,
+          payload: { chainAlias: PUBLIC_CHAIN_ALIAS }
+        }
+      })
+      .put.like({
+        action: {
+          type: actionTypes.CHAIN_LOADED,
+          payload: {
+            chainAlias: PUBLIC_CHAIN_ALIAS,
+            chainId: PUBLIC_CHAIN_ID
+          }
+        }
+      })
+      .put.like({
+        action: {
+          type: actionTypes.CHAIN_BLOCKING,
+          payload: { chainAlias: PUBLIC_CHAIN_ALIAS }
+        }
+      })
+      .put.like({
+        action: {
+          type: actionTypes.CHAIN_LOADING,
+          payload: { chainAlias: STATIC_CHAIN_ALIAS }
+        }
+      })
+      .put.like({
+        action: {
+          type: actionTypes.CHAIN_LOADED,
+          payload: {
+            chainAlias: STATIC_CHAIN_ALIAS,
+            chainId: STATIC_CHAIN_ID
+          }
+        }
+      })
+      .put.like({
+        action: {
+          type: actionTypes.CHAIN_BLOCKING,
+          payload: { chainAlias: STATIC_CHAIN_ALIAS }
+        }
+      })
+      .not.put.like({
+        action: {
+          type: actionTypes.CHAIN_ERROR
+        }
+      })
+      .run()
+  })
+
+  it('If private chain ID in local storage, load the chain', async () => {
+    const contextWithChainIdInLocalStorage = mockContext({
+      dataStoreContent: privateChainIdInLocalStorage
+    })
+    const { rootSaga } = middlewareSagas(contextWithChainIdInLocalStorage)
+    await expectSaga(rootSaga)
+      .withReducer(reducers)
+      .dispatch(
+        // We need the public chain ID
```

Comment:
Comment provides no context beyond the dispatch call. Why do we need the chain ID?